### PR TITLE
[Feature][Connector-V2] Support typesense connector

### PR DESCRIPTION
**Pull request by zhangshenghang**
### Purpose of this pull request

#7191

Support Typesense Connector

@hailin0  @Hisoka-X

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-typesense-e2e/src/test/java/org/apache/seatunnel/e2e/connector/typesense/TypesenseIT.java [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+import java.util.stream.Stream;
+
+@DisabledOnContainer(
+        value = {
+            TestContainerId.FLINK_1_13,
+            TestContainerId.FLINK_1_14,
+            TestContainerId.FLINK_1_15
+        },
+        type = {EngineType.SEATUNNEL, EngineType.SPARK},
+        disabledReason = "Test only one engine for first change")
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-typesense-e2e/src/test/java/org/apache/seatunnel/e2e/connector/typesense/TypesenseIT.java]: Disabled all engine?

**Comment by Hisoka-X** [comment]
Thanks @zhangshenghang !

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-typesense-e2e/src/test/java/org/apache/seatunnel/e2e/connector/typesense/TypesenseIT.java (hunk 1 above)
Comment (zhangshenghang) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-typesense-e2e/src/test/java/org/apache/seatunnel/e2e/connector/typesense/TypesenseIT.java]: > Disabled all engine?

sry ,  I blocked it during testing and forgot to restore it. I am now deleting it

**Comment by hailin0** [comment]
good pr

File: seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            default:
+                builder.dataType(BasicType.STRING_TYPE);
+                break;
+        }
+        return builder.build();
+    }
+
+    @Override
+    public BasicTypeDefine<TypesenseType> reconvert(Column column) {
+        return null;
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java]: null ?

**Comment by Hisoka-X** [comment]
Please add label too. https://github.com/apache/seatunnel/blob/c6f627fa38d29f83bccb3e1fb86e8962b64dad4e/.github/workflows/labeler/label-scope-conf.yml#L255

File: seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java (hunk 2 above)
Comment (zhangshenghang) [reply seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java]: @Hisoka-X 
This method is currently not used in the code. Do I still need to write in this situation? Do you have any good suggestions.

**Comment by zhangshenghang** [comment]
> Please add label too.
> 
> https://github.com/apache/seatunnel/blob/c6f627fa38d29f83bccb3e1fb86e8962b64dad4e/.github/workflows/labeler/label-scope-conf.yml#L255

already added

File: seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java (hunk 2 above)
Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java]: throw new UnsupportedException?

Comment (zhangshenghang) [reply seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java]: modify after:
```java
@Override
    public BasicTypeDefine<TypesenseType> reconvert(Column column) {
        throw new UnsupportedOperationException("Unsupported operation");
    }
```

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @zhangshenghang !

**Review by hailin0 — APPROVED** [review]

**Pull request merged**